### PR TITLE
Refactor and sol

### PR DESCRIPTION
**Pull request by Hanlee0707**
1. Refactored JS part out from index.html 
2. integrated webpack into the application so that we can import node.js libraries, though I am not sure if you would prefer to not have that dependency.  
3. Implemented solana domain resolver

### CODE REVIEW

File: src/queryexecutor.js [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  const hashQuery = hashParams.get("q");
+  if (hashQuery) {
+    return hashQuery.split(" ");
+  }
+  // Otherwise, there is no command set
+  return ["homepage"];
+}
+
+// Top 100 CoinGecko coins. Hardcoded to provide fast lookups.
+const COIN_GECKO_SYMBOL_MAP = {
```
Comment (zkhalapyan) [inline src/queryexecutor.js:208]: nit, we could extract this out

File: src/queryexecutor.js [hunk 2]
```diff
@@ -0,0 +1,350 @@
+const {AddressResolver} = require('./addressresolver.js');
+
+class QueryExecutor {
```
Comment (zkhalapyan) [inline src/queryexecutor.js:3]: does this need to be a class?

File: src/addressresolver.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+function getCoinTypeForHash_(hash) {
+  if (hash.startsWith('0x') || hash.includes('.eth')) {
+    return COIN_TYPE.ETHEREUM;
+  }
+  return COIN_TYPE.SOLANA;
+}
+
+function solScanURL_(path, env) {
```
Comment (zkhalapyan) [inline src/addressresolver.js:54]: is the _ postfix to indicate it's private? I have seen the prefix version of it, `_privateFunction()`, never seen this one

File: package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+  "devDependencies": {
+    "webpack": "^5.67.0",
+    "webpack-cli": "^4.9.1"
+  },
+  "name": "wagmify.me",
+  "description": "The tool makes it faster to navigate to common crypto sites — instead of going to CoinGeiko.ccom and searching for olympus, you can just type in `p ohm` and the tool will automatically redirect you.",
+  "version": "1.0.0",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (zkhalapyan) [inline package.json:10]: could be fun to add a test scaffold

**Review by zkhalapyan — COMMENTED** [review]
Nothing blocking, but I do think we should move to TS as soon as possible; especially now that we have NPM setup. TS's typing helps to resolve a lot of bugs during development time.

File: .gitignore [hunk 5]
```diff
@@ -1 +1,3 @@
 .env
```
Comment (zkhalapyan) [inline .gitignore:1]: What are you storing in env?

**Review by zkhalapyan — DISMISSED** [review]

File: src/queryexecutor.js (hunk 1 above)
Comment (Hanlee0707) [reply src/queryexecutor.js:208]: yeah I had whole another changes that refactored granular stuff out but did not know if it was ok to take dependency on web pack, we can do it after

File: src/queryexecutor.js (hunk 2 above)
Comment (Hanlee0707) [reply src/queryexecutor.js:3]: not really but no reason for it to not be

File: src/addressresolver.js (hunk 3 above)
Comment (Hanlee0707) [reply src/addressresolver.js:54]: It's a google thing https://google.github.io/styleguide/jsguide.html#jsdoc-visibility-annotations we can do either way

File: package.json (hunk 4 above)
Comment (Hanlee0707) [reply package.json:10]: down the road maybe

File: .gitignore (hunk 5 above)
Comment (Hanlee0707) [reply .gitignore:1]: I was storing some settings I thought was required... but I figured another way out. Can remove this

File: src/solanadomainconverter.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  async toWalletAddress() {
+    const { owner } = await NameRegistryState.retrieve(
+      new Connection(clusterApiUrl(this.env)),
+      await getNameAccountKey(
+        await getHashedName(this.domain.replace('.sol', '')),
+        null,
+        SOL_TLD_AUTHORITY,
+      )
+    );
+    return owner.toString();
```
Comment (kunalmodi) [inline src/solanadomainconverter.js:32]: <3

File: package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{
+  "devDependencies": {
+    "webpack": "^5.67.0",
+    "webpack-cli": "^4.9.1"
+  },
+  "name": "wagmify.me",
+  "description": "The tool makes it faster to navigate to common crypto sites — instead of going to CoinGeiko.ccom and searching for olympus, you can just type in `p ohm` and the tool will automatically redirect you.",
+  "version": "1.0.0",
+  "scripts": {
```
Comment (kunalmodi) [inline package.json:9]: Is there a command to build the main.js? / does it make sense to have our deploy pipeline build this or should we just compile main.js and commit it?

**Review by kunalmodi — APPROVED** [review]

**Comment by Hanlee0707** [comment]
ahh that's right, thanks for pointing that out Kunal.  The command should be 

npx webpack --config webpack.config.js

not sure if we need to add this to the deploy pipeline before we push these changes out.

**Pull request merged**